### ISSUE TITLE
TUI build is broken (v1.15)

### ISSUE THREAD

**Issue by sikmir** [skyjake/lagrange#571]
On macOS:
```
/opt/local/var/macports/build/_Users_sikmir_Projects_macports-ports_net_lagrange/lagrange/work/v1.15.0/src/main.c:70:17: error: use of undeclared identifier 'SDL_HINT_WINDOWS_DPI_AWARENESS'
    SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "permonitor");
                ^
1 error generated.
make[2]: *** [CMakeFiles/app.dir/src/main.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/opt/local/var/macports/build/_Users_sikmir_Projects_macports-ports_net_lagrange/lagrange/work/v1.15.0/src/app.c:732:21: error: incompatible pointer types assigning to 'iMainWindow *' (aka 'struct Impl_MainWindow *') from 'iMainOrExtraWindow *' (aka 'struct Impl_Window *') [-Werror,-Wincompatible-pointer-types]
                win = d->window;
                    ^ ~~~~~~~~~
```

On Linux:
```
lagrange> /tmp/nix-build-lagrange-1.15.0.drv-0/source/src/main.c:70:17: error: use of undeclared identifier 'SDL_HINT_WINDOWS_DPI_AWARENESS'
lagrange>     SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "permonitor");
lagrange>                 ^
lagrange> 1 error generated.
lagrange> make[2]: *** [CMakeFiles/app.dir/build.make:76: CMakeFiles/app.dir/src/main.c.o] Error 1
lagrange> make[2]: *** Waiting for unfinished jobs....
lagrange> /tmp/nix-build-lagrange-1.15.0.drv-0/source/src/app.c:732:21: error: incompatible pointer types assigning to 'iMainWindow *' (aka 'struct Impl_MainWindow *') from 'iMainOrExtraWindow *' (aka 'struct Impl_Window *') [-Werror,-Wincompatible-pointer-types]
lagrange>                 win = d->window;
lagrange>                     ^ ~~~~~~~~~
```

GUI build is OK at the same time.

SDL2 - 2.24.2 / 2.26.2
sealcurses - 2022-05-18 (417d77d790ede990b4c149f21c58fd13b8f273cc)

**Comment by skyjake** [comment]
Yeah, the TUI build has not yet been updated for the changes in v1.15.

**Comment by m040601** [comment]
yeap, can confirm.

Same thing happened to me on Archlinux.

```
[  1%] Building C object CMakeFiles/app.dir/src/main.c.o
/dev/shm/n2/lagrange/src/main.c: In function ‘main’:
/dev/shm/n2/lagrange/src/main.c:70:17: error: ‘SDL_HINT_WINDOWS_DPI_AWARENESS’ undeclared (first
use in this function)
   70 |     SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "permonitor");
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/dev/shm/n2/lagrange/src/main.c:70:17: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [CMakeFiles/app.dir/build.make:76: CMakeFiles/app.dir/src/main.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/app.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```


Although there is no explicit reference in the README to Archlinux (only Mac and Ubuntu ...), I know this should have worked because I have sucessfuly done this before in May 2022 and used clagrange. The "dependencies" for archlinux are very similar to ubuntu and are all Arch Linux official packages. 


I used this procedure:

1. git clone


```
$ git clone --recursive --branch release https://git.skyjake.fi/gemini/lagrange
```

2. edit "build-tui.sh"

Change

```
INSTALL_PREFIX="/usr/local"
```

to something temporary like

```
INSTALL_PREFIX="/dev/shm/temporary-clagrange/"
```

I dont want to "pollute"  my /usr/local with custom made installations or hacks. This is the job for my Linux distro package manager to handle and track all dependencies that I stuff to my system. 

Unfortunatelly there is (currently) no official Arch package for lagrange. Only an  AUR "recipe" PKGBUILD , https://aur.archlinux.org/packages/lagrange. It caters to "lagrange" not "clagrange"



3. So, I then ran build-tui.sh

```
./build-tui.sh
```

As expected it detected that I dont have that "sealcurses" thing,

```
'lib/sealcurses' not found. Clone with Git? [Yn]
```

That part went Ok.

It then went to the next part where I got the aforementioned "SDL_HINT_WINDOWS_DPI_AWARENESS" error.




Another wish of mine, not a request, or by any means a demand are the precompiled binaries for clagrange.

I'm not a programmer or developer, so I cant estimate if that would be too much work for skyjake. Maybe make use of github workflows ? Dont know.

Releasing pre compiled binaries for TUI clagrange just like for GUI lagrange would also be very usefull. Especially for those little underpowerd ARM machines, raspberry, chromebooks etc. 





EXTRA:

I mostly use CLI gemini clients. I know them all.  So TUI clagrange has good competition there. Fortunately the "market" is not "dying" just like for normal web browsers. But still hope to continue to see this TUI "clagrange" well supported and easy to build. There are some very nice user intereface patterns from GUI lagrange that could  (?) be replicated in TUI clagrange. Like that "Next set of home row key links". Awesome.

As a die-hard cli/tui apps fan I dont even use a desktop most of the time. 
But I make an exception for GUI Lagrange. I sometimes even boot into X11 just to use it.

It's the only well thought, well made GUI I've seen in the last years. Thank you for all your awesome work.

**Comment by skyjake** [comment]
The TUI build is fixed in v1.15.2. Don't forget to update to the latest sealcurses.

@m040601 The plan is to make prebuilt binaries of clagrange eventually. I haven't decided exactly how, perhaps using AppImage.

> "Next set of home row key links"

This, along with pretty much all the features of the GUI version (that are keyboard-accessible), are in the TUI build as well. They mostly use the same/similar keybindings.

I would still label the TUI port as "experimental", though, because it lacks some affordances that are common in TUI apps. For instance, there could be a context-dependent nano-like keyboard shortcut list in the bottom of the screen. Some of the interactions are also somewhat clunky for TUI as they have been designed for GUI. For example, a cursor-based text selection and link activation mode would be useful.
